packages/tools/src/comparison/compare-runs.ts

```diff
@@ -0,0 +1,257 @@
+async function unzip(zippedPath: string, unzippedPath: string) {
+  const readStream = fs.createReadStream(zippedPath)
+  const unzipStream = unzipper.Extract({
+    path: unzippedPath,
+    forceStream: true,
+  })
+  await new Promise((resolve, reject) => {
+    readStream.on('error', reject)
+    unzipStream.on('error', reject)
+    unzipStream.on('close', resolve)
+    readStream.pipe(unzipStream)
+  })
+}
+
+function compareModelOutputs(modelOutputs: output.ModelOutput[]) {
+  logger.info('Comparing run outputs')
+  assert.deepEqual(
+    modelOutputs[0].metadata,
+    modelOutputs[1].metadata,
+    'Model output metadata do not match'
+  )
+  assert.equal(
+    modelOutputs[0].time.t0,
+    modelOutputs[1].time.t0,
+    'Starting timestamps do not match'
+  )
+  assert.sameOrderedMembers(
+    modelOutputs[0].time.extent,
+    modelOutputs[1].time.extent,
+    'Simulation extents do not match'
+  )
+  assert.sameOrderedMembers(
+    modelOutputs[0].time.timestamps,
+    modelOutputs[1].time.timestamps,
+    'Simulation timestamps do not match'
+  )
+
+  const sum = (xs: number[]) => xs.reduce((a, b) => a + b, 0)
+  const last = (xs: number[]) => xs[xs.length - 1]
+  const totalDeaths = modelOutputs.map(o => sum(o.aggregate.metrics.incDeath))
+
+  const CUMULATIVE_DEATHS_DIFFERENCE_THRESHOLD = 1000
+  const CUMULATIVE_CASES_DIFFERENCE_THRESHOLD = 1000
+  assert.approximately(
+    totalDeaths[1],
+    totalDeaths[0],
+    CUMULATIVE_DEATHS_DIFFERENCE_THRESHOLD,
+    'Total deaths vary'
+  )
+
+  const totalCases = modelOutputs
+    .map(o => o.aggregate.metrics)
+    .map(metrics =>
+      [
+        metrics.cumMild,
+        metrics.cumILI,
+        metrics.cumSARI,
+        metrics.cumCritical,
+      ].map(last)
+    )
+    .map(sum)
+  assert.approximately(
+    totalCases[1],
+    totalCases[0],
+    CUMULATIVE_CASES_DIFFERENCE_THRESHOLD,
+    'Total cases vary'
+  )
+
+  // TODO compare other metrics
+}
+
+suite('comparing runs', async () => {
+  let tmpDir: string
+  suiteSetup(async () => {
+    tmpDir = await fs.promises.mkdtemp(
+      path.join(__dirname, 'model-runner-validation')
+    )
+    logger.info('Working directory: %s', tmpDir)
+  })
+  suiteTeardown(async () => {
+    await fs.promises.rmdir(tmpDir, { recursive: true })
+  })
+
+  let inputZips: string[]
+  test('process zip files to be compared', async () => {
+    const baseResultsZip = process.env.BASE_RESULTS_ZIP
+    assert.isDefined(
+      baseResultsZip,
+      'Set the environment variable BASE_RESULTS_ZIP to the path of a zipped results folder.'
+    )
+    const comparisonResultsZip = process.env.COMPARISON_RESULTS_ZIP
+    assert.isDefined(
+      comparisonResultsZip,
+      'Set the environment variable COMPARISON_RESULTS_ZIP to the path of a zipped results folder.'
+    )
+    inputZips = [baseResultsZip, comparisonResultsZip]
+
+    // Unzip the given artifacts.
+    for (let i = 0; i < 2; i++) {
```

Comment:
Some of these artifacts are pretty big. Can we do this in parallel?

Reply:
Good idea.